### PR TITLE
Add merge_ids_op

### PR DESCRIPTION
**Pull request by jacquesqiao**
project: https://github.com/PaddlePaddle/Paddle/issues/10868

dist lookup table返回的结果需要merge在一起，之前用的concat是不对的，会打乱结果的顺序，merge_ids参考原始Id的顺序才能把结果正确的merge在一起，同时也把lod信息拷贝给merge的结果。

### CODE REVIEW

File: paddle/fluid/framework/executor.cc [hunk 1]
```diff
@@ -330,8 +330,12 @@ void Executor::RunPreparedContext(ExecutorPrepareContext* ctx, Scope* scope,
   }
 
   for (auto& op : ctx->ops_) {
-    VLOG(3) << place_ << " " << op->DebugStringEx(local_scope);
+    VLOG(4) << place_ << " " << op->DebugStringEx(local_scope);
```
Comment (Yancey1989) [inline paddle/fluid/framework/executor.cc]: Can we keep the log level before `op run` and after?

File: paddle/fluid/operators/merge_ids_op.cc [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        "X",
+        "(LoDTensors) multi input tensor with shape{batch_num, N}, N is the "
+        "size of embedding table")
+        .AsDuplicable();
+    AddOutput("Out", "(LoDTensor) The merged outputs of the input tensors.");
+
+    AddComment(R"DOC(
+Merge multi LoDTensor's into one according to Ids's shard num.
+The values in the input LoDTensor are lookuped from the output of split_ids_op
+
```
Comment (Yancey1989) [inline paddle/fluid/operators/merge_ids_op.cc]: How about adding some comments to explain why X0 is mapped to index 3 and 6.

File: paddle/fluid/operators/merge_ids_op.h [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+      VLOG(3) << "only one shard, we can copy the data directly";
+      TensorCopy(*x_tensors[0], place, out);
+    } else {
+      std::vector<int> in_indexs(shard_num, 0);
+      auto *out_data = out->mutable_data<T>(
+          framework::make_ddim({batch_size, embedding_size}), place);
+      // copy data from ins[shard_num] to out.
+      for (int i = 0; i < ids_dims[0]; ++i) {
+        int64_t id = ids[i];
+        size_t shard_id = static_cast<size_t>(id) % shard_num;
```
Comment (typhoonzero) [inline paddle/fluid/operators/merge_ids_op.h]: Hash method is currently hardcoded, just a message for notice.

Comment (jacquesqiao) [reply paddle/fluid/operators/merge_ids_op.h]: yes, will be an independent module in the future.

File: paddle/fluid/operators/merge_ids_op.cc (hunk 2 above)
Comment (jacquesqiao) [reply paddle/fluid/operators/merge_ids_op.cc]: updated

**Pull request merged**